api/src/test/java/org/pmiops/workbench/actionaudit/auditors/SumoLogicAuditorTest.java

```diff
@@ -0,0 +1,216 @@
+    verify(mockActionAuditService).send(eventsCaptor.capture());
+    Collection<ActionAuditEvent> events = eventsCaptor.getValue();
+
+    // Some of the properties should be nulled out, since we can't identify the target workspace
+    // for the egress event.
+    assertThat(events.stream().map(event -> event.getAgentId()).collect(Collectors.toSet()))
+        .containsExactly(0);
+    assertThat(events.stream().map(event -> event.getAgentEmailMaybe()).collect(Collectors.toSet()))
+        .containsExactly((Object[]) null);
+    assertThat(events.stream().map(event -> event.getTargetIdMaybe()).collect(Collectors.toSet()))
+        .containsExactly((Object[]) null);
+
+    // We expect to see an audit event row with a comment describing the issue encountered when
+    // trying to handle the high-egress message.
+    assertThat(
+            events.stream()
+                .filter(
+                    event ->
+                        event.getTargetPropertyMaybe()
+                            == EgressEventCommentTargetProperty.COMMENT.getPropertyName())
+                .map(event -> event.getNewValueMaybe())
+                .findFirst()
+                .get())
+        .contains("Failed to find workspace");
+  }
+
+  @Test
+  public void testFailedParsing() {
+    // When the inbound request parsing fails, an event is logged at the system agent.
+    when(mockWorkspaceService.getByNamespace(WORKSPACE_NAMESPACE)).thenReturn(null);
+    sumoLogicAuditor.fireFailedToParseEgressEvent(new EgressEventRequest().eventsJsonArray("asdf"));
+    verify(mockActionAuditService).send(eventsCaptor.capture());
+    Collection<ActionAuditEvent> events = eventsCaptor.getValue();
+
+    assertThat(
+            events.stream()
+                .filter(
+                    event ->
+                        event.getTargetPropertyMaybe()
+                            == EgressEventCommentTargetProperty.COMMENT.getPropertyName())
+                .map(event -> event.getNewValueMaybe())
+                .findFirst()
+                .get())
+        .contains("Failed to parse egress event");
+  }
+
+  @Test
+  public void testBadApiKey() {
+    // When the inbound request parsing fails, an event is logged at the system agent.
+    when(mockWorkspaceService.getByNamespace(WORKSPACE_NAMESPACE)).thenReturn(null);
+    sumoLogicAuditor.fireBadApiKeyEgressEvent("ASDF", new EgressEventRequest());
+    verify(mockActionAuditService).send(eventsCaptor.capture());
+    Collection<ActionAuditEvent> events = eventsCaptor.getValue();
+
+    assertThat(
+            events.stream()
+                .filter(
+                    event ->
+                        event.getTargetPropertyMaybe()
+                            == EgressEventCommentTargetProperty.COMMENT.getPropertyName())
+                .map(event -> event.getNewValueMaybe())
+                .findFirst()
+                .get())
+        .contains("Received bad API key");
+  }
+
+  @TestConfiguration
```

Comment:
nit: Oh, I'm used to seeing these at the top.

Reply:
Yeah I don't know why these ended up down here... moved back up.